Update docs for useAnimatedStyle (#5153)
software-mansion/react-native-reanimated: docs/docs/core/useAnimatedStyle.mdx

`useAnimatedStyle` returns an animated style object which has to be passed to the `style` property of an [Animated component](/docs/fundamentals/glossary#animated-component) that you want to animate.

> Types of `useAnimatedStyle` are made so they mimic behavior of `StyleSheet` styles as much as possible - `updater` callback return value should just look like a regular style object - with the exception of using shared values.

Reviewer: ```suggestion
`useAnimatedStyle` mimics the behavior of `StyleSheet` as much as possible. `updater` callback returns a value that looks like a regular style object in which you can also use shared values.
```
A bit simplified. You can rewrite this sentence in another way if you want but try to keep it simple 🙏

> :::caution

Reviewer: ```suggestion

```

We had a tendency to overuse the `:::caution` and `:::note` tags in the past. I'd recommend avoiding using it. If everything is marked as important, nothing is.

> For better compliance with custom components, `useAnimatedStyle` return value type might seem like it can be appended to a non-animated component - but this will result in an error. Always use `useAnimatedStyle` return value with animated components only.

Reviewer: Please rewrite it in active voice and simplify it. It's definitely written too in a too complicated fashion than it needs to be

## Remarks

> - Mutating shared values in `useAnimatedStyle`'s callback is considered a bad practice. While it might work most of the time, it can lead to undefined behavior and infinite loops.

Reviewer: ```suggestion
- Mutating shared values in `useAnimatedStyle`'s callback is an undefined behavior which may lead to infinite loops.
```

Simplified to the point

| ✅      | ✅  | ✅  |

</div>

> ## Advanced

Reviewer: Can we put this in Remarks so we won't have a couple sections that serve the same purpose?

```suggestion

```

> - The callback passed to the `useAnimatedStyle` is first run on the [JavaScript thread](/docs/fundamentals/glossary#javascript-thread) and immediately after on the [UI thread](/docs/fundamentals/glossary#ui-thread). This might cause an error had you expected UI thread execution only. To avoid this, you can use the `global._WORKLET` variable to check if the code is running on the UI thread:

Reviewer: good job 👍
